src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.cs

```diff
@@ -69,58 +69,83 @@ private static void Execute(CompilationData compilationData, ImmutableArray<Bind
         private sealed record CompilationData
         {
             public bool LanguageVersionIsSupported { get; }
-            public KnownTypeData? TypeData { get; }
+            public KnownTypeSymbols? TypeSymbols { get; }
 
             public CompilationData(CSharpCompilation compilation)
             {
                 LanguageVersionIsSupported = compilation.LanguageVersion >= LanguageVersion.CSharp11;
                 if (LanguageVersionIsSupported)
                 {
-                    TypeData = new KnownTypeData(compilation);
+                    TypeSymbols = new KnownTypeSymbols(compilation);
                 }
             }
         }
 
-        private sealed record KnownTypeData
+        private sealed record KnownTypeSymbols
         {
-            public INamedTypeSymbol SymbolForGenericIList { get; }
-            public INamedTypeSymbol SymbolForICollection { get; }
-            public INamedTypeSymbol SymbolForIEnumerable { get; }
-            public INamedTypeSymbol SymbolForString { get; }
-
-            public INamedTypeSymbol? SymbolForConfigurationKeyNameAttribute { get; }
-            public INamedTypeSymbol? SymbolForDictionary { get; }
-            public INamedTypeSymbol? SymbolForGenericIDictionary { get; }
-            public INamedTypeSymbol? SymbolForHashSet { get; }
-            public INamedTypeSymbol? SymbolForIConfiguration { get; }
-            public INamedTypeSymbol? SymbolForIConfigurationSection { get; }
-            public INamedTypeSymbol? SymbolForIDictionary { get; }
-            public INamedTypeSymbol? SymbolForIServiceCollection { get; }
-            public INamedTypeSymbol? SymbolForISet { get; }
-            public INamedTypeSymbol? SymbolForList { get; }
-
-            public KnownTypeData(CSharpCompilation compilation)
+            public INamedTypeSymbol GenericIList { get; }
+            public INamedTypeSymbol ICollection { get; }
+            public INamedTypeSymbol IEnumerable { get; }
+            public INamedTypeSymbol String { get; }
+
+            public INamedTypeSymbol? CultureInfo { get; }
+            public INamedTypeSymbol? DateOnly { get; }
+            public INamedTypeSymbol? DateTimeOffset { get; }
+            public INamedTypeSymbol? Guid { get; }
+            public INamedTypeSymbol? Half { get; }
+            public INamedTypeSymbol? Int128 { get; }
```

Comment:
>Int128 [](http://example.com/codeflow?start=37&length=6)

What about the other numeric types? like int, uint, long, ulong...etc?

Reply:
Those types are ["special"](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.specialtype?view=roslyn-dotnet) in Roslyn APIs, so we can do type checks based on `ITypeSymbol.SpecialType`.